File: src/main/java/org/railwaystations/rsapi/core/model/Country.java
package org.railwaystations.rsapi.core.model;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Value;

import java.util.ArrayList;
import java.util.List;

@Value
@Builder
public class Country {

    @EqualsAndHashCode.Include
    String code;

    String name;

    @Builder.Default
    String email = "info@railway-stations.org";

    @Builder.Default
    String twitterTags = "@Bahnhofsoma, #dbHackathon, #dbOpendata, #Bahnhofsfoto, @khgdrn";

    String timetableUrlTemplate;

    License overrideLicense;

    boolean active;

    @Getter
    List<ProviderApp> providerApps = new ArrayList<>();

}

File: src/main/kotlin/org/railwaystations/rsapi/core/model/Country.kt
package org.railwaystations.rsapi.core.model

data class Country(
    val code: String,
    val name: String,
    val email: String = "info@railway-stations.org",
    val timetableUrlTemplate: String? = null,
    val overrideLicense: License? = null,
    val active: Boolean = false,
    val providerApps: MutableList<ProviderApp> = mutableListOf(),
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Country

        return code == other.code
    }

    override fun hashCode(): Int {
        return code.hashCode()
    }
}

